ci/jenkins/utils.sh

```diff
@@ -0,0 +1,36 @@
+#!/bin/bash
+
+# Copyright 2023 Antrea Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+function check_and_upgrade_golang() {
+    echo "====== Checking current Golang version ======"
+    current_version=$(/usr/local/go/bin/go version | grep -oP "go\d+\.\d+(\.\d+)?")
+    project_golang_version=$(go mod edit -json | jq -r .Go)
+    if [ "$current_version" = "go$project_golang_version" ]; then
+        echo "====== Golang version on the testbed is $current_version ======"
+        echo "====== No need to upgrade ======"
+    else
+        echo "====== Upgrading golang version from "$current_version" to "go$project_golang_version" ======"
+        install_golang "go$project_golang_version"
+    fi
+}
+
+function install_golang() {
+    golang_versin=$1
+    echo "====== Downloading Golang $project_golang_version ======"
+    curl https://dl.google.com/go/${golang_versin}.linux-amd64.tar.gz -o /tmp/${golang_versin}.linux-amd64.tar.gz
```

Comment:
are you sure this is correct?
The link https://dl.google.com/go/go1.21.linux-amd64.tar.gz doesn't exist

Reply:
The `go.mod` file does not contain the patch number, for example `go 1.21`, but it is required by `https://go.dev/dl`. So we need to add the patch number ourselves, I'll fix it.